src/screens/AppScreen.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  faBriefcase,
+  faCircleQuestion,
+  faMap,
+  faQrcode,
+  faUser,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
+import React from "react";
+import { useRecoilState } from "recoil";
+import { currentUserInfo } from "../recoil/atom";
+import BagScreen from "./BagScreen";
+import LogInScreen from "./LogInScreen";
+import MapScreen from "./MapScreen";
+import ProfileScreen from "./ProfileScreen";
+import QuestNavigator from "./QuestNavigator";
+import ScannerScreen from "./ScannerScreen";
+
+export type RootStackParamList = {
+  Map: undefined;
+  Scanner: undefined;
+  QuestNavigator: undefined;
+  Bag: undefined;
+  Profile: undefined;
+};
+
+const Tab = createBottomTabNavigator<RootStackParamList>();
+
+const AppScreen = () => {
+  const [user] = useRecoilState(currentUserInfo);
+
```

Comment:
Maybe not use inline conditional rendering? More difficult to read in my opinion.
```suggestion
  if (!user.username) {
    return <LogInScreen />;
}
```